orttraining/orttraining/test/python/orttraining_run_bert_pretrain.py

```diff
@@ -0,0 +1,531 @@
+    )
+
+    gradient_accumulation_steps: Optional[int] = field(
+        default=1,
+        metadata={"help": "Number of updates steps to accumualte before performing a backward/update pass."}
+    )
+
+    fp16: bool = field(
+        default=False,
+        metadata={"help": "Whether to use 16-bit float precision instead of 32-bit."}
+    )
+
+    loss_scale: Optional[float] = field(
+        default=0.0,
+        metadata={"help": "Loss scaling, positive power of 2 values can improve fp16 convergence."}
+    )
+
+    log_freq: Optional[float] = field(
+        default=1.0,
+        metadata={"help": "frequency of logging loss."}
+    )
+
+    checkpoint_activations: bool = field(
+        default=False,
+        metadata={"help": "Whether to use gradient checkpointing."}
+    )
+
+    resume_from_checkpoint: bool = field(
+        default=False,
+        metadata={"help": "Whether to resume training from checkpoint."}
+    )
+
+    resume_step: Optional[int] = field(
+        default=-1,
+        metadata={"help": "Step to resume training from."}
+    )
+
+    num_steps_per_checkpoint: Optional[int] = field(
+        default=100,
+        metadata={"help": "Number of update steps until a model checkpoint is saved to disk."}
+    )
+
+    phase2: bool = field(
+        default=False,
+        metadata={"help": "Whether to train with seq len 512."}
+    )
+
+    allreduce_post_accumulation: bool = field(
+        default=False,
+        metadata={"help": "Whether to do allreduces during gradient accumulation steps."}
+    )
+
+    allreduce_post_accumulation_fp16: bool = field(
+        default=False,
+        metadata={"help": "Whether to do fp16 allreduce post accumulation."}
+    )
+
+    accumulate_into_fp16: bool = field(
+        default=False,
+        metadata={"help": "Whether to use fp16 gradient accumulators."}
+    )
+
+    phase1_end_step: Optional[int] = field(
+        default=7038,
+        metadata={"help": "Whether to use fp16 gradient accumulators."}
+    )
+
+    tensorboard_dir: Optional[str] = field(
+        default=None,
+    )
+
+    schedule: Optional[str] = field(
+        default='warmup_poly',
+    )
+
+    # this argument is test specific. to run a full bert model will take too long to run. instead, we reduce
+    # number of hidden layers so that it can show convergence to an extend to help detect any regression.
+    force_num_hidden_layers: Optional[int] = field(
+        default=None,
+        metadata={"help": "Whether to use fp16 gradient accumulators."}
+    )
+
+def setup_training(args):
+
+    assert (torch.cuda.is_available())
+
+    if args.local_rank == -1:
+        args.local_rank = 0
+
+    print("args.local_rank: ", args.local_rank)
+    torch.cuda.set_device(args.local_rank)
+    device = torch.device("cuda", args.local_rank)
+    args.n_gpu = 1
+
+    from onnxruntime.capi._pybind_state import set_cuda_device_id
+    set_cuda_device_id(args.local_rank)
```

Comment:
```suggestion
```
This is old API. Setting `ORTTrainerOptions.device.id` already calls this function in the frontend